### DISCUSSION TITLE
New to OpenFAST

### DISCUSSION THREAD

**Discussion by mowayjan** [OpenFAST/openfast#1672] [Q&A] [answered]
Hello all, I am new to OpenFAST. I have downloaded the latest version and watched some videos online on how to create a simulation for an FOWT. I have successfully simulated the 5MW turbine in the r-test folder as well as the IEA-15-240-RWT Semisubmersible and was able to visualize the turbine using ParaView. However, I was not able to activate the control, for example the torque control, and was giving me errors. Is there a guide on how to simulate and control a 15MW or 20MW FOWT?
Another issue was with the fst and dat files on the IEA-15-240 RWT, they were missing many variables/lines that I have added from another example. Is there an official site where I can download the files?
A last issue I faced is with the SFunction in Simulink/MATLAB, it wasn't working with the 2018 version of MATLAB. How can I make it work? Do I have to download the source and compile it? I don't have Visual Studio.
Thank you for the answers and your time.

**Comment by jjonkman** [comment, marked as answer]
Dear @mowayjan,

Regarding controller-related errors; what error(s) are you receiving?

Regarding the IEA Wind 15-MW reference wind turbine models, my guess is that the repository is based on a different version of OpenFAST than you have downloaded.  You can either download a version of OpenFAST that is consistent with the IEA Wind 15-MW repository or upgrade the model from the IEA Wind 15-MW repository to the version of OpenFAST you have downloaded.  The changes to the input file(s) with each release of OpenFAST are documented here: https://openfast.readthedocs.io/en/main/source/user/api_change.html.

Regarding the OpenFAST S-Function not working; what error(s) are you receiving?

Best regards,

**Reply by mowayjan** [reply]
Dear Jason @jjonkman,

Thank you very much for your reply and sorry about my late reply.

I have decided to work on the NREL 5-MW FOWT with OC4 Semi-submersible platform. Maybe when I finish/validate my work, I will try on the IEA 15 MW model.

My work is in the intelligent model-free control of FOWT operating in above nominal wind (Region III). I am working on controlling the generator speed (to stabilize the power) and I am comparing the results with the GSPI controller developed in your published technical report about the definition of the turbine and the platform. I am using OpenFAST 3.4.1 with MATLAB 2028b version.

I have applied on the NREL site for an account to be able to post questions in the forum, but I am still waiting for the authorization.

Thank you very much for your support and I hope I will be able to share with you my results soon.

Regards,
Michel

**Reply by jjonkman** [reply]
Dear @mowayjan ,

Thank you for your interest in an account on the NREL forums. If you wish to have an account on the forums, you must first register by clicking the Sign Up button. Please also make sure you have read the community rules and instructions for creating a forum account: https://forums.nrel.gov/t/requesting-access-to-our-forum/2639/5. I don't recall receiving the e-mail required for me to approve your forum account.

Best regards,
